pkg/tfgen/ids.go

```diff
@@ -0,0 +1,43 @@
+package tfgen
+
+import (
+	"fmt"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+)
+
+func checkIDProperties(sink diag.Sink, spec schema.PackageSpec, info tfbridge.ProviderInfo) error {
+	errors := 0
+
+	info.P.ResourcesMap().Range(func(rname string, resource shim.Resource) bool {
+		_, gotID := resource.Schema().GetOk("id")
+		if gotID {
+			return true
+		}
```

Comment:
Should we do anything if the `id` field is not a string or is marked as secret?

Reply:
This might make the check overly strict and annoying as it blocks provider compilation. I can try though. Let me write a strict check and trigger a downstream provider build+test from here.